### DISCUSSION TITLE
What is the reasoning behind lowering the getItemOffersBatch request rate?

### DISCUSSION THREAD

**Discussion by callumb123** [amzn/selling-partner-api-models#3586] [General]
I have just seen the post "Starting May 1, 2023, the rate for the getItemOffersBatch operation within the Product Pricing API will be lowered" which states that the getItemOffersBatch is going to be made FIVE times slower with less than 1 months notice.

This seems extreme and unwarranted, what was the thought process behind this and why have we been given less than 1 months notice?

Only being able to request an endpoint once every 10 seconds that is crucial to mine and many other's applications is crazy. Is the 'max items' being increased from 20 to offset some of the damage this will cause?

@jevoniuk @alex-rupprecht @hudalexa @andiec1

**Comment by gokigoks** [comment]
Same boat as OP. Amazon, please reconsider this change.

**Comment by jim-foley** [comment]
Agreed, too drastic of a change in too short of time. Hoping Amazon comes back on this.

**Comment by chadbst** [comment]
Agree as well.  Five times slower!!  Please Amazon reconsider this.

**Comment by txphi592** [comment]
Same issue, this is a huge problem for us. Please reconsider!

**Comment by jamesp26** [comment]
Agreed. In the early days of migration over to SPAPI we were informed that the goal was for longer term improvements to these end points. Please reconsider this change.

**Comment by kallen** [comment]
Agreed.  This change is too drastic and would be detrimental to our use case.  Please reconsider this change.

**Comment by sungolivia** [comment]
hey folks, thanks for the feedback, I have taken this to our product team and will provide an update as soon as I hear any! In the meantime, if you can expand on how this affects you aside from the obvious, it would really help drive this further!

**Reply by txphi592** [reply]
Thanks for your help! We have a bulk use-case where Amazon sellers use our app to process 10s or 100s of thousands of UPC/EANs (per day) and processing speed is absolutely critical. We provide services to thousands of sellers so this change is somewhat devastating and will impact us and many sellers negatively.

**Reply by callumb123** [reply]
Our use case is pretty much the same as @txphi592, our clients bulk scan lists of UPCs to determine products to sell. We also have processes to keep the pricing of these products up to date. This current change will mean the initial scan will take 5 times longe as well as significantly reducing the accuracy of the data we are able to provide as we will only be able to update 20% of the products we can currently.

As has been mentioned above, we were under the impression that SP API would perform the same as or better than MWS but this change is a significant step backwards. With only 8 days to go till this change is expected to take place I really hope something can be done quickly.

**Comment by callumb123** [comment]
@sungolivia - Are you able to provide any update on this? How likely is it to be resolved before the current go-live date (this week)?

**Comment by callumb123** [comment]
@sungolivia - I hate to be impatient, but sadly the extremely short notice we have been given for this change leaves me with little choice! And I am aware I am unlikely to get a response over the weekend.

Could you please let us know what is happening here? I need to know if I have to tell my clients that our service will be 5 times slower come Monday

**Comment by txphi592** [comment]
Looks like this got [pushed back to July 10th](https://github.com/amzn/selling-partner-api-docs/discussions/3109) which is nice, I guess, but does not address the real problem. Dropping the speed 5x will still be the same problem in July as it was in May. I'm hoping this extra time enables reasonable minds to reconsider this headscratcher completely? @sungolivia

**Reply by callumb123** [reply]
Couldn't agree more! The short timescale was the secondary concern, the real concern is the change itself. Making the API worse than MWS and significantly impacting several businesses does not seem like a good idea.

**Comment by sungolivia** [comment]
Hey guys couldn't provide an update until we had this out amzn/selling-partner-api-docs#3109 
We are still pushing internally for this and appreciate all the feedback. Definitely helped with the detailed use case and clear impact on you guys helped to get this at least. 
I understand this is not what you were looking for but please know that this issue is still being raised.

However, our internal team was wondering the following and if yall can help them understand by answering this, it would be awesome!
1. Can you migrate calls to [getlistingoffers](https://developer-docs.amazon.com/sp-api/docs/product-pricing-api-v0-reference#getlistingoffers)
2. how many ASINs are you tracking
3. how often are you updating pricing

**Reply by chadbst** [reply]
Going to add my voice here as well.  I use getItemOffersBatch to make purchasing decisions on Amazon.  I monitor prices of books, buy them then and resell them on Amazon as well as other channels.  I purchase around $100K worth of items per year on Amazon(give or take depdending on the year).  This decision to lower the throttle this dramatically has me looking to change careers - it's that bad!  I am in the early stages of going to exploring grad school because I don't think my business will survive the impact of this.

**Reply by chadbst** [reply]
By the way sungolivia - thank you for being an advocate within Amazon regarding this.  I really appreciate your effort.

**Reply by chadbst** [reply]
Sorry, I didn't see your three questions until now.  

1.) No, I cannot migrate to getlistingoffers.  Getlistingoffers is getting the offer information of a sku that is already in your inventory.
2.) ~70K
3.) approx every 2 to 3 hours.

**Reply by callumb123** [reply]
Thanks for getting back to us @sungolivia, it's great to hear this is still being looked at! Here are my answers to your questions:

Can you migrate calls to getlistingoffers - No, our application is used to help our clients source products so the products we want to get pricing for are not in our client's inventory.

how many ASINs are you tracking - This changes per client and depends how large the UPC lists they upload are. Some lists may be 50k long which currently would take around 1.4 hours, with this change it would now take 7 hours for our clients to run a single sheet. Some of our clients may upload over 500k UPCs per day which would no longer be possible with this change. We also have 250,000 ASINs we track separately which we update about every 8 hours.

how often are you updating pricing - Our own tracked ASINs are every 8 hours, however our clients would be every time they upload a UPC list which can vary drastically day to day.

**Reply by nfzv** [reply]
1. Unfortunately, no
2. ~300k
3. every 2-3 hours

**Reply by jim-foley** [reply]
Hi @sungolivia 

1. No, this is only to lookup sellerSKU, not an ASIN. Most of our clients use the data for sourcing the listings they want to sell on - not already sell. 
2. Same boat as several other users who have commented, our application hosts nearly a thousand sellers. It all varies and while some sellers need to max out their requests, it doesn't mean they are full throttle 24/7. However, the ability to get the data when they need it at the current rates is crucial to their business. I don't know to what degree but I can tell you for certain this change would translate to less offers from sellers -> less items sold -> less money in Amazons pocket; it would be a loss across the board. 
3. Same as above. 

Thank you.

**Reply by turbopiranha** [reply]
Hi @sungolivia , 
Same as other responders, we have a bulk product analysis software and our clients check the profitability and the competition. Some clients run our software 24/7 with either huge bulk files, or they queue the smaller files and get the results back to back. This is a huge step back for SP-API and even worse than MWS! We will definitely have a difficulty to explain this issue to our clients and we have to to change the core principles of the software.

**Comment by txphi592** [comment]
Hi @sungolivia 

1) No we cannot migrate calls to getlistingoffers since the app is used to source NEW products that our clients do not currently have in inventory. 
2) Same answer as @callumb123 + high volume. We have thousand+ customers processing millions of products per day in aggregate. This change would absolutely crush these sellers since this is a bulk use-case and processing speed is critical. In other words, this change would require 5 days for an Amazon seller to analyze what currently takes 1 day... They will be dead in the water!  I was hoping/expecting that SP-API would allow for FASTER processing speed vs MWS not 5X slower...what are they thinking?
3)  Basically 24/7 as our customers process/analyze bulk lists.

**Comment by callumb123** [comment]
@sungolivia - Any update on this yet?

We are all of course desperately hoping this change isn't going to go ahead at all, but if there are going to be any changes made it would be good to have much more notice once the final decision has been made.

There have been far too many occasions with SP API where I haven't been given a final decision until hours before a change, I'm trying to avoid that happening here.

**Comment by psop819058** [comment]
Proposal:  Please consider increasing the max ASINs per request from 20 to 40, and then instead of reducing the throttling limit from .5 to .1 / second, reduce it from .5 to .2 / second.

The current throttling rate is effectively 10 ASINs / second, and on July 10 it will go to 2 ASINs / second.  The proposal would make it effectively 8 ASINs / second:

* 8 ASINs / second is probably a tolerable reduction in speed (20%) for most features developers are offering sellers
* going from 20 to 40 ASINs per request shouldn't make the response size too large
* it reduces by 60% the number of requests hitting Amazon's SP-API servers

**Reply by turbopiranha** [reply]
That's a very reasonable suggestion.

**Comment by callumb123** [comment]
Without any update from Amazon here I raised a new support case. Support can be useless so I would take this with a pinch of salt but they have said "For now, the lowering of the getItemOffersBatch rate limits will be maintained and scheduled to July 10th, 2023"

While I'm hopeful @sungolivia will come back with better news, it seems the current plan is to continue with this devastating change.

**Reply by sungolivia** [reply]
@rugved1991  for visibility!

**Reply by callumb123** [reply]
@sungolivia @rugved1991 Any update? We are right down to the wire as always

**Reply by gokigoks** [reply]
Plenty of people who use this endpoint are anxious for an update. And I think I speak for many.

**Comment by callumb123** [comment]
After the change did not go ahead on the 10th as planned, or for many days after that, I was hopeful Amazon had heard our plea and were reconsidering their options. However, it appears they have decided to go ahead with this now as 'x-amzn-ratelimit-limit' is now returning as 0.1

@sungolivia @rugved1991 Please tell us this is still being investigated and that some form of middle ground can be achieved!

**Comment by jim-foley** [comment]
@sungolivia @rugved1991 Many users here have spent time offering feedback and presenting their use case at your request but we have got back little in return. While there may only be 11 participants here, we are the spokespeople for thousands of Amazon sellers who use our software. Our inboxes have been flooded this week with complaints and we strongly feel the 0.1 rate limit is unreasonable and out of place compared to other endpoints. We're just asking for something more reasonable that will not disturb the workflow of Amazon sellers who rely on this data to sell more product. Please come back to us and acknowledge we're not wasting our time here. We appreciate you as an advocate if this is something you need to push to your team.

**Reply by jamesp26** [reply]
Agree with jim-foley. Last few days has been a nightmare trying to work around this issue as it was impossible to know what the reality was going to look like before it changed.

**Comment by fr3shn3ss** [comment]
We scan physical books during our 8 hour shift and we have multiple employees

Now we have to wait 10 seconds for a few asins? We used to batch the requests every 2 seconds but now that is not possible. 

Our inventory is piling up and it's been a week since I created a support case to increase the limit. WAITING

Such a big change and they didn't even send a single email about it. Just in Announcements which I missed

**Comment by txphi592** [comment]
Follow up- 

The cost and disruption to our customers as a result of this change has been quite dramatic. Dealing with throttling issues associated with getItemOffersBatch has caused us to reduce processing speeds by more than 3x. Our customers are not happy since they are now 3x less productive which has a direct flow-through to their efficiency and the efficiency of a large chunk of the Amazon seller community. 

Further, this change creates a void in the market which will be filled with alternative 3rd party API providers such as Keepa, Rainforest, etc. These providers gather data running massive scraping operations which drain Amazon resources. These APIs exist primarily due to the limitations of SP-API, there is a direct correlation. 

I wonder if the decision makers on this Amazon product team really understood the impact of these decisions? I would be happy to contribute seller partner feedback if they are ever interested..

I really hope Amazon rethinks this decision as it has been quite a head scratcher, to put it nicely. @sungolivia @rugved1991

**Comment by callumb123** [comment]
Bumping - In the hope that someone from Amazon may one day decide to listen to developer feedback and provide us with an update

@sungolivia @rugved1991

**Comment by fr3shn3ss** [comment]
After a month of fighting I got my limit increased back to 0.5/sec. Give it a shot. However, we got it increase only for a single seller. Not sure if they would do it for multiple sellers

**Reply by callumb123** [reply]
Thanks for letting me know @fr3shn3ss, sadly we have hundreds of clients so need an application level 'increase'.

**Reply by fr3shn3ss** [reply]
I assume you share api limits across users. don't you? If you do, is that still not enough?
